### ISSUE TITLE
Augmented Reality / Virtual Reality Demonstration of Weather Real-Time Data APIs

### ISSUE THREAD

**Issue by bcdevex** [bcgov/data-stream#2]
<p>This issue has been auto-generated to facilitate questions about
<a href="https://bcdevexchange.org/opportunities/cwu/opp-augmented-reality---virtual-reality-demonstration-of-weather-real-time-data-apis"> a paid opportunity</a> that has just been posted on the
<a href="https://bcdevexchange.org">BCDevExchange</a>.</p>

<p>To learn more or apply, <a href="https://bcdevexchange.org/opportunities/cwu/opp-augmented-reality---virtual-reality-demonstration-of-weather-real-time-data-apis">visit the opportunity page</a>.
The opportunity closes on <b>Friday, February 15th, 2019 at 16:00 PST</b>.</p>

**Comment by berashith** [comment]
Roughly what quantity of data should be transferred via API for the demonstration, and how often should it be updating within the VR/AR application?

**Comment by berashith** [comment]
Does the Phase 2 / Phase 3 AR/VR product need to be multiplatform, or can it be targeted, e.g. Vive specific?
Is a multiplatform solution preferable, or merely something that allows a compelling demonstration in Phase 3?

**Comment by berashith** [comment]
Regarding proposal evaluation criterion one: are we intended to demonstrate we already have a VR application geared to this use-case, or is it intended to demonstrate that we can rapidly create finished VR products / platforms.

**Comment by berashith** [comment]
Are we able to access the current Data API to build a functional prototype as part of the proposal?

**Comment by berashith** [comment]
Does Phase 1 item 2 (Testing APIs connections) indicate a comprehensive test of scalability, or a proof-of-concept test of successfully accessing API data?

**Comment by berashith** [comment]
Regarding Phase 2, is it possible to get more information regarding the desired demonstration? 
This would allow a much better tailored app to be created in the time allotted. Suggested use-cases or items beyond the listed 3 that you would like to see demonstrated would help direct the development.

**Comment by ShacharWeis** [comment]
I have a question about the timeline. The work is slated to start on Feb 20th, and phase 3 requires demonstrating a working VR on March 11. Is that March 11 of this year, or next year? 3 weeks to complete phase 1 and 2 doesn't seem feasible.

**Comment by berashith** [comment]
> I have a question about the timeline. The work is slated to start on Feb 20th, and phase 3 requires demonstrating a working VR application in BCTech summit on March 11. Is that March 11 of this year, or next year? 3 weeks to complete phase 1 and 2 doesn't seem feasible.

Hi SacharWeis - this is for this year's BCTech Summit: https://www.bctechsummit.ca/

**Comment by ShacharWeis** [comment]
> > I have a question about the timeline. The work is slated to start on Feb 20th, and phase 3 requires demonstrating a working VR application in BCTech summit on March 11. Is that March 11 of this year, or next year? 3 weeks to complete phase 1 and 2 doesn't seem feasible.
> 
> Hi SacharWeis - this is for this year's BCTech Summit: https://www.bctechsummit.ca/

Are you sure? A $70k project to be completed in 3 weeks? I find that hard to believe.

**Comment by berashith** [comment]
We're planning on having a proof-of-concept prototype ready for the Application deadline. It's doable, but will be a lot of work very quickly (I think that plus the specialization required is the reason for the pricetag) though more information will let us all hone our proposals.

The dates mentioned match this year's BC Tech Summit, and the dates for next year's will be different.

**Comment by ShacharWeis** [comment]
That's a good point about the dates. I guess we'll wait for an official answer. I agree that's it's possible to build a functioning prototype within 3 weeks, we've certainly done it before. I'm just not used to seeing these kind of time frames in a government project :)

**Comment by berashith** [comment]
What is the current state of the API? Is there another weather API whose structure it is most similar to?

**Comment by berashith** [comment]
What scale / resolution is the data available by API? Is this city level data like OpenWeatherMap, or street-level data like ClimaCell?

**Comment by AlexanderRitchie** [comment]
Hi everyone. I'm Alexander Ritchie, the contact for this project. I'll do my best to provide responses to your questions.
Please feel free to ask others as the week goes forward.

**Comment by AlexanderRitchie** [comment]
> This issue has been auto-generated to facilitate questions about [ a paid opportunity](https://bcdevexchange.org/opportunities/cwu/opp-augmented-reality---virtual-reality-demonstration-of-weather-real-time-data-apis) that has just been posted on the [BCDevExchange](https://bcdevexchange.org).
> 
> To learn more or apply, [visit the opportunity page](https://bcdevexchange.org/opportunities/cwu/opp-augmented-reality---virtual-reality-demonstration-of-weather-real-time-data-apis). The opportunity closes on Friday, February 15th, 2019 at 16:00 PST.

**Comment by AlexanderRitchie** [comment]
> Roughly what quantity of data should be transferred via API for the demonstration, and how often should it be updating within the VR/AR application?

We have a rough idea, but expect to work with the successful vendor to fully define the data types, quantity, volume, and timing of data to be delivered within the API. We are planning for up to 10 points of data in a burst and 10 polygons on a pre-determined time based schedule.

**Comment by AlexanderRitchie** [comment]
> Does the Phase 2 / Phase 3 AR/VR product need to be multiplatform, or can it be targeted, e.g. Vive specific?
> Is a multiplatform solution preferable, or merely something that allows a compelling demonstration in Phase 3?

The AR/VR product doesn’t need to be multiplatform or a specific platform. Our intention is to provide the simulated weather event data API(s) and the vendor will develop the code in the open to interact with the API(s). The code of the AR/VR product platform is in the hands of the vendor as that is simply for user demonstration purposes. Our shared interest in the API delivering data through ‘API callbacks/webhooks’ and the code that interacts with the API.

**Comment by AlexanderRitchie** [comment]
> Regarding proposal evaluation criterion one: are we intended to demonstrate we already have a VR application geared to this use-case, or is it intended to demonstrate that we can rapidly create finished VR products / platforms?

We prefer and expect a vendor that has VR applications geared toward the use-case of weather data in order to meet the timelines of tech summit demonstration; our interest is not in the rapid development of VR applications; we are looking to demonstrate the value of real-time data and ‘API callbacks/webhooks’ to deliver event driven data, using the VR weather use-case to prove out the data sharing approach.

**Comment by AlexanderRitchie** [comment]
> Are we able to access the current Data API (or its documentation) to build a functional prototype as part of the proposal?

No, the API is under development and will need to be finalized with the selected vendor. We are not looking for a functional prototype to be submitted as part of the proposal.

**Comment by AlexanderRitchie** [comment]
> Does Phase 1 item 2 (Testing APIs connections) indicate a comprehensive test of scalability / stress test, or a proof-of-concept test of successfully accessing API data?

Testing API connections is to prove the successful access of the API; we are not looking to test scalability at this time.

**Comment by AlexanderRitchie** [comment]
> Regarding Phase 2, is it possible to get more information regarding the desired demonstration?
> This would allow a much better tailored app to be created in the time allotted. Suggested use-cases or items beyond the listed 3 that you would like to see demonstrated would help direct the development.

We are looking for ideas from vendors and experts at demonstrating the use of weather-data in AR/VR applications, rather than prescribing the nature of the demonstration use-case. We have ideas of the weather event-data that we will be exposing through the API, but need to work with the vendor to finalize those details.

**Comment by AlexanderRitchie** [comment]
> I have a question about the timeline. The work is slated to start on Feb 20th, and phase 3 requires demonstrating a working VR application in BCTech summit on March 11. Is that March 11 of this year, or next year? 3 weeks to complete phase 1 and 2 is very tight.

We are keen to show the delivery of a proof-of-concept demonstration at the upcoming #BCTech Summit, March 11-13, 2019. We understand the timelines are tight, but are keen to more forward with this exciting work to prove the value of delivering real-time event data through API(s)

**Comment by AlexanderRitchie** [comment]
> > I have a question about the timeline. The work is slated to start on Feb 20th, and phase 3 requires demonstrating a working VR application in BCTech summit on March 11. Is that March 11 of this year, or next year? 3 weeks to complete phase 1 and 2 doesn't seem feasible.
> 
> Hi SacharWeis - this is for this year's BCTech Summit: https://www.bctechsummit.ca/

Yes, confirmed the intent is for the 2019 #BCTech Summit, March 11-13.

**Comment by AlexanderRitchie** [comment]
> What is the current state of the API? Is there another weather API whose structure it is most similar to?

The API is currently under development. We are building out the API ‘callback/webhook’ capabilities. As mentioned, we have some ideas of the data to provide and the likely data structure but need to finalize that with the successful vendor. There is not another weather API that we can reference at this time.

**Comment by AlexanderRitchie** [comment]
> What scale / resolution is the data available by API? Is this city level data like OpenWeatherMap, or street-level data like ClimaCell?

We are developing with the context of the provincial scale, but will work with the successful vendor to define a specific region or area of interest to create the simulated data for the demonstration purposes. We are thinking resolution at street-level or larger (more detailed) scale.

**Comment by AlexanderRitchie** [comment]
> > This issue has been auto-generated to facilitate questions about [ a paid opportunity](https://bcdevexchange.org/opportunities/cwu/opp-augmented-reality---virtual-reality-demonstration-of-weather-real-time-data-apis) that has just been posted on the [BCDevExchange](https://bcdevexchange.org).
> > To learn more or apply, [visit the opportunity page](https://bcdevexchange.org/opportunities/cwu/opp-augmented-reality---virtual-reality-demonstration-of-weather-real-time-data-apis). The opportunity closes on Friday, February 15th, 2019 at 16:00 PST.

**Comment by ShacharWeis** [comment]
For the event in March, do you have a VR headset that you wish to use? If yes, which one? If no, should the successful vendor bring their own VR equipment?

**Comment by AlexanderRitchie** [comment]
We are looking for the successful vendor to bring all AR/VR products and related services and capabilities to the project and the demonstration. We’ll provide the weather data API but need the vendor to bring everything else.

**Comment by AlexanderRitchie** [comment]
> For the event in March, do you have a VR headset that you wish to use? If yes, which one? If no, should the successful vendor bring their own VR equipment?

We are looking for the successful vendor to bring all AR/VR products and related services and capabilities to the project and the demonstration. We’ll provide the weather data API but need the vendor to bring everything else.

**Comment by nikg** [comment]
What use-cases are targeted for the dataset / API in its final state? Considering it's a real-time data API, who will be ingesting it and for what purpose? Examples can help to shape the demo / POC ideas & delivery for the event.

**Comment by simiansim** [comment]
> We have a rough idea, but expect to work with the successful vendor to fully define the data types, quantity, volume, and timing of data to be delivered within the API. We are planning for up to 10 points of data in a burst and 10 polygons on a pre-determined time based schedule.

@AlexanderRitchie Thank you, your answers have been very helpful. I would love to further discuss the API design and web-hook scalability patterns during the project, but I think we are hoping to understand what is available in the data to understand if this fits with our available visualization patterns/expectations. It is not important to be able to provide a finished API spec at this time, but we would like to know:
1) what data points might be available (i.e. temperature, wind speed/direction, humidity, dew point, etc.)
2) what example polygons, for which purpose, are available. (I am assuming this is geospatial, possibly boundaries for regions or weather patterns/fronts)
3) what time intervals are involved in the sourcing real data? (and for demo this might be manipulated for effect)

**Comment by AlexanderRitchie** [comment]
> What use-cases are targeted for the dataset / API in its final state? Considering it's a real-time data API, who will be ingesting it and for what purpose? Examples can help to shape the demo / POC ideas & delivery for the event.

We are looking for ideas from vendors and experts at demonstrating the use of weather-data in AR/VR applications, rather than prescribing the nature of the demonstration use-case. We have ideas of the weather event-data that we will be exposing through the API, but need to work with the vendor to finalize those details.

**Comment by countable** [comment]
Hello, my team has the following questions regarding this opportunity:

1. What is the purpose and motivation behind developing this weather API?
2. Who is the long term intended user of the API and what are their expected use cases?
3. What things would constitute a successful launch of the weather API?

Thanks for your time and providing all the great answers so far!

**Comment by AlexanderRitchie** [comment]
> Hello, my team has the following questions regarding this opportunity:
> 
> 1. What is the purpose and motivation behind developing this weather API?
> 2. What things would constitute a successful launch of the weather API?
> 3. Is an entry-level headset such as Google Cardboard considered ok for this demo? If so, may we use some of the budget to give away Google Cardboard headsets as part of the demo?
> 
> Thanks for your time and providing all the great answers so far!
> 
> *edit, removed a question that was answered already.

1) Purpose and motivation of developing the API(s) is to further understand both the development and consumption of real-time event based APIs to deliver streaming data, particularly the use of API callbacks/webhooks to provide interactive data APIs. We are keen to better understand the value of opening up various datasets and the technical patterns to best make the data open.
2) For this pilot project, successful launch will be the weather API(s) available to deliver data for the initial demonstration and functional pattern of making real-time event data available through APIs
3) We are not prescribing the type of VR/AR products we expect, but are evaluating submissions on the team having a production ready product and platform to bring to the project.

**Comment by mark-a-wilson** [comment]
Hello interested applicants .... there was confusion around the instruction to submit a PDF proposal .... 
Please do sign in and submit a proposal through the BCDevExchange application. When you start a proposal and save it in DRAFT, you should then see an Attachments tab appear in the proposal where you can upload a PDF and/or other files.

Let me know here if you have problems.

**Comment by berashith** [comment]
Would costs aside from physical assets be covered for presentation at BCTech Summit? 
Would costs associated in-person sessions in Victoria be covered? 
How much expected in-person work would there be?

**Comment by AlexanderRitchie** [comment]
> Would costs aside from physical assets be covered for presentation at BCTech Summit?
> Would costs associated in-person sessions in Victoria be covered?
> How much expected in-person work would there be?

1) There won't be any additional expenses covered beyond the project value for attendance at the summit.
2) There won't be any additional expenses covered beyond the project value for travelling for in-person sessions in Victoria
3) We anticipate the need for multiple in-person sessions to ensure the teams are collaborating effectively over the course of the project; the successful vendor will need to ensure that they can meet in-person requirements to meet the objectives and timelines of the project; specific interaction details can be worked out with the successful vendor and the internal project team as part of kicking-off the work.

**Comment by ShacharWeis** [comment]
Good luck to everyone that applied!

**Comment by bcdevex** [comment]
This opportunity has been assigned

**Comment by ShacharWeis** [comment]
I would love to know how many applications you got. Can you share that information?

Thanks!

**Comment by AlexanderRitchie** [comment]
Hi everyone, thanks for all of your robust submissions. There were many great proposals. In total we received 11 submissions and the winning proposal was from LlamaZOO. We are happy to provide feedback to unsuccessful submissions. Please contact me at alexander.ritchie@gov.bc.ca if you would like some feedback.

**Comment by ShacharWeis** [comment]
I am very curious to know if this project got to phase 2 or 3. I don't see any VR code in the repository. Is there a demo running at #BCTech Summit?

**Comment by AlexanderRitchie** [comment]
Thanks for checking. Yes we have been busily completing this project and will be demoing the final result tomorrow and Wednesday at the #BCTech Summit. The API calling code from LlamaZOO will be posted shortly, we had never expected the VR code to be developed in the open.

**Comment by ShacharWeis** [comment]
Good news :)

Will there be a video or pictures of the VR app? I'd love to see it.
